### PR TITLE
MM-55589: Throw warn logs for MySQL < 8

### PR DESCRIPTION
**Pull request by agnivade**
We just throw a warn log for now. Support will be completely
removed next ESR.

https://mattermost.atlassian.net/browse/MM-55589
```release-note
MySQL 5.7 is at EOL. We recommend all customers to upgrade to atleast 8.x. For now, we are logging a warning. From version 9.5 onwards, which is the next ESR, we will stop supporting 5.7 altogether.
```

### CODE REVIEW

**Comment by agnivade** [comment]
https://github.com/mattermost/mattermost/pull/25638 sent.

**Review by streamer45 — APPROVED** [review]

**Comment by agnivade** [comment]
Gentle ping @nab-77

File: server/channels/store/sqlstore/store.go [hunk 1]
```diff
@@ -1231,6 +1232,9 @@ func (ss *SqlStore) ensureMinimumDBVersion(ver string) (bool, error) {
 			return false, fmt.Errorf("cannot parse MySQL DB version: %w", err2)
 		}
 		intVer := majorVer*1000 + minorVer*100 + patchVer
+		if intVer < minimumRecommendedMySQLVersion {
+			mlog.Warn("The currently installed MySQL version is at EOL. Please upgrade to a higher version.", mlog.Int("current_version", intVer), mlog.Int("minimum_recommended_version", minimumRecommendedMySQLVersion))
+		}
 		if intVer < minimumRequiredMySQLVersion {
 			return false, fmt.Errorf("minimum MySQL version requirements not met. Found: %s, Wanted: %s", versionString(intVer, *ss.settings.DriverName), versionString(minimumRequiredMySQLVersion, *ss.settings.DriverName))
```
Comment (nab-77) [inline server/channels/store/sqlstore/store.go:1239]:
```suggestion
			return false, fmt.Errorf("Minimum MySQL version requirements not met. Found: %s, Wanted: %s", versionString(intVer, *ss.settings.DriverName), versionString(minimumRequiredMySQLVersion, *ss.settings.DriverName))
```

File: server/channels/store/sqlstore/store.go [hunk 2]
```diff
@@ -1231,6 +1232,9 @@ func (ss *SqlStore) ensureMinimumDBVersion(ver string) (bool, error) {
 			return false, fmt.Errorf("cannot parse MySQL DB version: %w", err2)
 		}
 		intVer := majorVer*1000 + minorVer*100 + patchVer
+		if intVer < minimumRecommendedMySQLVersion {
+			mlog.Warn("The currently installed MySQL version is at EOL. Please upgrade to a higher version.", mlog.Int("current_version", intVer), mlog.Int("minimum_recommended_version", minimumRecommendedMySQLVersion))
```
Comment (nab-77) [inline server/channels/store/sqlstore/store.go:1236]:
```suggestion
			mlog.Warn("The MySQL version being used is EOL. Please upgrade to a later version.", mlog.Int("current_version", intVer), mlog.Int("minimum_recommended_version", minimumRecommendedMySQLVersion))
```

**Review by nab-77 — APPROVED** [review]

**Comment by amyblais** [comment]
@agnivade This missed the v9.4 feature complete. Is this for v9.4 or v9.5?

**Comment by agnivade** [comment]
Ah sorry, we got a bit delayed in making the decision. It will be good to place this in 9.4, as in 9.5, we are planning to block MySQL 5.7 completely.

**Pull request merged**